src/drivers/codeModel.ts

```diff
@@ -7,7 +7,7 @@ import * as api from "vscode-cmake-tools";
 // public API.
 export type CodeModelConfiguration = api.CodeModel.Configuration;
 export type CodeModelContent = api.CodeModel.Content;
-export type CodeModelFileGroup = api.CodeModel.FileGroup;
+export type CodeModelFileGroup = api.CodeModel.FileGroup & { frameworks?: { path: string }[] };
```

Comment:
Why couldn't we share code and use `FrameworkMetadata` here? It seems like they should be the same type, but I don't have full understanding of the FileApi should if I'm wrong, I'd love to learn why.

Reply:
Added a TODO comment. I'm leaving the actual intersection type as-is for now (no `FrameworkMetadata`) because while they share names right now, there's no reason why they have to. `FrameworkMetadata` represents the serialization format from CMake, while this type is simply however we choose to represent it internally. TypeScript will still catch type errors if there's a issue because we never cast between these types.